### PR TITLE
 Sequelize v4 support #4

### PR DESCRIPTION
**Pull request by kos984**
Sequelize v4 support #4

### CODE REVIEW

**Comment by hbj** [comment]
@kos984 thanks for the PR. Can you please rebase it on the `develop` branch?

**Comment by kos984** [comment]
@hbj done

File: README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ To activate the plugin on specific models, call the plugin on the models:
 var sequelizeTransforms = require('sequelize-transforms');
 
 var Model = sequelize.define('Model', { /* model definition */ });
-sequelizeTransforms(Model);
+sequelizeTransforms(sequelize, Model);
```
Comment (hbj) [inline README.md:33]: Not needed, please roll back.

File: README.md [hunk 2]
```diff
@@ -62,7 +62,7 @@ The plugin comes with the following predefined transforms:
 It is possible to override predefined transforms or add your own by passing an object as the second argument:
 
 ```js
-sequelizeTransforms(sequelize, {
+sequelizeTransforms(sequelize, null, {
```
Comment (hbj) [inline README.md:65]: Not needed, please roll back.

File: index.js [hunk 3]
```diff
@@ -14,8 +14,8 @@ var defaultTransforms = {
   }
 };
 
-function init(target, transforms) {
-  if (target instanceof Sequelize.Model) {
+function init(Sequelize, target, transforms) {
```
Comment (hbj) [inline index.js:17]: Not needed, please roll back.

File: index.js [hunk 4]
```diff
@@ -57,8 +57,8 @@ function init(target, transforms) {
       target.refreshAttributes();
   }
   else {
-    target.afterDefine(function(Model) {
-      init(Model, transforms);
+    Sequelize.afterDefine(function(Model) {
```
Comment (hbj) [inline index.js:60]: Not needed, please roll back.

File: index.js [hunk 5]
```diff
@@ -57,8 +57,8 @@ function init(target, transforms) {
...
+      init(Sequelize, Model, transforms);
```
Comment (hbj) [inline index.js:61]: Not needed, please roll back.

File: test/index.spec.js [hunk 6]
```diff
@@ -94,7 +94,7 @@ describe('Sequelize transforms', function() {
   });
 
   it('should run custom transforms', function() {
-    sequelizeTransforms(sequelize, {
+    sequelizeTransforms(sequelize, null,{
```
Comment (hbj) [inline test/index.spec.js:97]: Not needed, please roll back.

File: test/index.spec.js [hunk 7]
```diff
@@ -122,7 +122,7 @@ describe('Sequelize transforms', function() {
 
     assert.strictEqual(instance.trim, '  Test String  ');
 
-    sequelizeTransforms(Model);
+    sequelizeTransforms(Sequelize, Model);
```
Comment (hbj) [inline test/index.spec.js:125]: Not needed, please roll back.

**Review by hbj — CHANGES_REQUESTED** [review]
Actually the only needed change seems to be replacing `target instanceof Sequelize.Model` with `Sequelize.Model.isPrototypeOf(target)`, so please roll back all other changes (added corresponding code comments).

I suggest also to add the same Sequelize version as peer dependency.

File: index.js [hunk 8]
```diff
@@ -14,8 +14,8 @@ var defaultTransforms = {
   }
 };
 
-function init(target, transforms) {
-  if (target instanceof Sequelize.Model) {
+function init(Sequelize, target, transforms) {
+  if (Sequelize.Model.isPrototypeOf(target)) {
```
Comment (sameoldmadness) [inline index.js:18]: Will this work for v3 as well?

Comment (hbj) [reply index.js:18]: Good point @sameoldmadness , the first idea was actually not to keep compatibility with Sequelize v3 in v2, but it's actually easy to do if we keep both conditions. I'll just merge this PR and do the fixes myself.

**Pull request merged**